activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -189,7 +212,8 @@ def setup_transactional_fixtures
             if pool
               setup_shared_connection_pool
 
-              unless @fixture_connection_pools.include?(pool)
+              # Don't begin a transaction if we've already done so, or are not using them for this pool
+              unless @fixture_connection_pools.include?(pool) || transactional_tests_for_pool?(pool)
```

Comment:
I think this should be `!`... assuming that's true, we're probably too complicated for an `unless`, and should at least flip to an `if`.

I see some tests are failing at the moment: if this is why there's upset, cool! If not, we should make sure this is covered.

Reply:
Yeah, I missed the negation. Also just flipped to using an `if`.